components/raftstore/src/store/peer.rs

```diff
@@ -1684,13 +1692,12 @@ where
                         );
                     }
                 }
-                StateRole::Follower => {
+                StateRole::Follower | StateRole::Candidate | StateRole::PreCandidate => {
```

Comment:
Why?

Reply:
For the exit force leader case, it becomes a follower, and then `campaign()` makes it becomes a pre-candidate, so it needs to clean some states in candidate/pre-candidate state as well.

Reply:
But there is no lease in force leader state.